Run all lint tasks in TeamCity CI

aduth opened the pull request:
<!--
Link a related Github/Linear issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword. Consider using "part of" instead.
-->

## Proposed Changes

* Fixes lint tasks to ensure they're being run in TeamCity builds

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* As evidenced by the existing issues fixed by these changes, lint tasks are not very effective unless applied consistently, such as through CI
* Simplifies future effort toward adopting consistent Prettier / Stylelint tooling for SCSS code style enforcement

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Verify `yarn lint` passes.

Easier to review by hiding whitespace-only changes: https://github.com/Automattic/wp-calypso/pull/103396/files?w=1

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

aduth committed:
Fix unused-state-action-types invoked by NPM script

It's a bash script, it shouldn't be evaluated using Node.js

aduth committed:
Add missing lint tasks to TeamCity

aduth committed:
Fix existing indentation issues in stylesheets

aduth committed:
Fix more mixed indentation

youknowriad commented on .teamcity/_self/projects/WebApp.kt:
```diff
@@ -686,6 +686,7 @@ object CheckCodeStyleBranch : BuildType({
 			scriptContent = """
 				# In the future, we may add the stylelint cache here.
 				yarn run lint:css
+				yarn run lint:mixedindent
```
Why do we have separate commands for these instead of them being included in the previous lint command?

aduth replied:
> Why do we have separate commands for these instead of them being included in the previous lint command?

If the question is "Why don't we check mixed indentation as part of `lint:css`?", my opinion is: They should be the same command, and we shouldn't have a separate tool / script for `lint:mixedindent`, and all of this should ideally be part of the same Stylelint + Prettier setup. 

Historically (prior to #102784), `lint:mixedindent` used to be used for more than just SCSS files, but we now enforce JavaScript/TypeScript indentation through Prettier. 

There's a lot of existing issues with stylesheets that will need to be resolved to get them fully linted with Prettier. I see this pull request as a small step toward that effort, carrying forward `lint:mixedindent` intentionally as a temporary companion to our CSS linting.

aduth replied:
> Why do we have separate commands for these instead of them being included in the previous lint command?

Another potential interpretation here is why we have separate TeamCity steps for linting CSS and JavaScript. I'm not as familiar here, but there does seem to be some fragmentation that I'm having a hard time following. In particular, we have two separate ESLint checks[[1]](https://github.com/Automattic/wp-calypso/blob/df6a0a4a503ea2a7a02d8dba6262e1c9db27d9cf/.teamcity/settings.kts#L263-L270)[[2]](https://github.com/Automattic/wp-calypso/blob/df6a0a4a503ea2a7a02d8dba6262e1c9db27d9cf/.teamcity/_self/projects/WebApp.kt#L601-L682), and some of our linters [live at the top-level project](https://github.com/Automattic/wp-calypso/blob/df6a0a4a503ea2a7a02d8dba6262e1c9db27d9cf/.teamcity/settings.kts#L56) and [others at the WebApp ](https://github.com/Automattic/wp-calypso/blob/df6a0a4a503ea2a7a02d8dba6262e1c9db27d9cf/.teamcity/_self/projects/WebApp.kt#L20), and I'm not clear what the distinction is. Is the former only run on `trunk`? Do people get notified? I see current failures, for example.

aduth replied:
Okay, I think I understand better:

1. `settings.kts` linters are scheduled to daily on all files in trunk, and don't have any impact on individual branches
2. `WebApp.kts` runs on each branch. For ESLint, only files changed in that branch are analyzed, presumably to improve performance and allow incremental adoption of new ESLint rules ([example](https://github.com/Automattic/wp-calypso/pull/103078))

As far as it impacts this branch:

- My changes should be updated to edit `WebApp.kt` instead of `settings.kts` to ensure they block pull requests on failure. They're fast enough to not impede development turnaround time. (**Edit:** Updated in 6b7f0cc8d0e)

Separately, I question the value of a daily ESLint build if nobody's accountable to them or paying attention, evidenced by the current failures.

aduth committed:
Move code linters to WebApp project

See: https://github.com/Automattic/wp-calypso/pull/103396/files#r2087554831

aduth committed:
Drop redundant n flag in xargs

Avoid warning in CI:

"xargs: warning: options --max-args and --replace/-I/-i are mutually exclusive, ignoring previous --max-args value"

`-I` already implies processing one line at a time

aduth committed:
Use "find" instead of "grep" for CI compatibility
